Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -431,13 +416,12 @@ open class NavigationMapView: UIView {
                     try mapView.mapboxMap.style.addSource(arrowSource, id: NavigationMapView.SourceIdentifier.arrowSource)
                     arrowLayer.source = NavigationMapView.SourceIdentifier.arrowSource
                     
-                    if let puckLayer = puckLayerIdentifier, allLayerIds.contains(puckLayer) {
-                        try mapView.mapboxMap.style.addPersistentLayer(arrowLayer, layerPosition: .below(puckLayer))
-                    } else if mapView.mapboxMap.style.sourceExists(withId: NavigationMapView.LayerIdentifier.waypointCircleLayer) {
-                        try mapView.mapboxMap.style.addPersistentLayer(arrowLayer, layerPosition: .below(NavigationMapView.LayerIdentifier.waypointCircleLayer))
-                    } else {
-                        try mapView.mapboxMap.style.addPersistentLayer(arrowLayer)
+                    var layerPosition = layerPosition(for: NavigationMapView.LayerIdentifier.arrowLayer, route: route)
+                    if let roadLabelIdentifier = mapView.mapboxMap.style.allLayerIdentifiers.compactMap({ $0.id }).filter({ $0.contains("road-label") }).last,
```

Comment:
The arrow layers will be placed above the road name, but below the exit and road shield labels. If no road name label existed, the arrows will be placed below all symbol layers.